### DISCUSSION TITLE
identify unused columns in a Power BI .pbix file

### DISCUSSION THREAD

**Discussion by ChristopherEby** [DaxStudio/DaxStudio#808] [Q&A]
is there a way to identify unused columns in a Power BI .pbix file so that they can be removed to increase overall performance?

**Comment by marcosqlbi** [comment]
By removing a column you only improve the processing time and reduce the memory of the model. You do not improve the query performance of your reports.
There is DMV providing information about columns referenced in DAX expressions of the model:
SELECT * FROM $System.DISCOVER_CALC_DEPENDENCY

However, this does not discover columns used in Power BI reports.

The Power BI cleaner tool tracks the dependency of columns used in a Power BI report. 
https://www.thebiccountant.com/2021/08/23/power-bi-cleaner-gen2-now-covers-analysis-services-models-as-well/
However, if you publish a dataset, you are not able to know which other clients might use the columns of the model (even though you would analyze all the published Power BI reports, you would not be able to detect Excel files connected and/or Power BI Desktop models connected to the dataset.
